base/sparse/sparsematrix.jl

```diff
@@ -1390,7 +1390,77 @@ function one{T}(S::SparseMatrixCSC{T})
+    # those above: If A.colptr and/or A.rowval are shorter than their expected lengths, the
+    # copy!s might catch the error, whereas the former copy/resize! combinations won't. The
+    # lines above are simpler/cleaner though. Thoughts?
+    # Bcolptr = similar(A.colptr, A.n + 1)
+    # copy!(Bcolptr, 1, A.colptr, 1, A.n + 1)
+    # Browval = similar(A.rowval, nnz(A))
+    # copy!(Browval, 1, A.rowval, 1, nnz(A))
+end
+"""
+Returns a (dense) `SparseMatrixCSC` with `f(zero(eltype(A)))` stored in place of all unstored
+entries in `A` and `broadcast(f, A.nzval)` stored in place of all stored entries in `A`.
+"""
+function _broadcast_notzeropres{Tf}(f::Tf, A::SparseMatrixCSC)
+    nnzB = A.m * A.n
+    # Build structure
+    Bcolptr = similar(A.colptr, A.n + 1)
+    copy!(Bcolptr, 1:A.m:(nnzB + 1))
+    Browval = similar(A.rowval, nnzB)
+    for k in 1:A.m:(nnzB - A.m + 1)
+        copy!(Browval, k, 1:A.m)
+    end
+    # Populate values
+    Bnzval = fill(f(zero(eltype(A))), nnzB)
+    @inbounds for (j, jo) in zip(1:A.n, 0:A.m:(nnzB - 1)), k in nzrange(A, j)
+        Bnzval[jo + A.rowval[k]] = f(A.nzval[k])
+    end
+    # NOTE: Combining the fill call into the loop above to avoid multiple sweeps over /
+    # nonsequential access of Bnzval does not appear to improve performance
+    return SparseMatrixCSC(A.m, A.n, Bcolptr, Browval, Bnzval)
+end
+
+# Cover common broadcast operations involving a single sparse matrix and one or more
+# broadcast scalars.
+#
+# TODO: The minimal snippet below is not satisfying: A better solution would achieve
+# the same for (1) all broadcast scalar types (Base.Broadcast.containertype(x) == Any?) and
+# (2) any combination (number, order, type mixture) of broadcast scalars. One might achieve
+# (2) via a generated function. But how one might simultaneously and gracefully achieve (1)
+# eludes me. Thoughts much appreciated.
+#
+broadcast{Tf}(f::Tf, x::Union{Number,Bool}, A::SparseMatrixCSC) = broadcast(y -> f(x, y), A)
+broadcast{Tf}(f::Tf, A::SparseMatrixCSC, y::Union{Number,Bool}) = broadcast(x -> f(x, y), A)
+# NOTE: The following two method definitions work around #19096. These definitions should
+# be folded into the two preceding definitions on resolution of #19096.
+broadcast{Tf,T}(f::Tf, ::Type{T}, A::SparseMatrixCSC) = broadcast(y -> f(T, y), A)
+broadcast{Tf,T}(f::Tf, A::SparseMatrixCSC, ::Type{T}) = broadcast(x -> f(x, T), A)
+
+# TODO: Determine what to do about collapsing the nz2nz_z2z and nz2z_z2z classes. Either (1)
+# the generic unary sparse broadcast behavior should be nz2z_z2z instead of the nz2z_z2z
+# behavior presently above (this would be consistent with existing binary sparse broadcast
+# behavior); or (2) the generic unary broadcast behavior should be nz2nz_z2z as above, and the
+# nz2z_z2z specializations below should ultimately go away in favor of `broadcast`+`dropzeros!`
+# (this would not be consistent with existing binary sparse broadcast behavior).
 
 """
```

Comment:
Once the nz2z/nz2nz decision is made, most of the code below this line till `## Broadcasting kernels specialized for returning a SparseMatrixCSC` should go away.